provider/core/src/serde/mod.rs

```diff
@@ -2,38 +2,186 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-//! Collection of traits for providers that support serializing or deserializing data.
-//!
-//! ## Deserializing
+//! Provides the [`DeserializingBufferProvider`] wrapper, which deserializes data using Serde.
 //!
 //! Providers that produce opaque buffers that need to be deserialized into concrete data structs,
-//! such as `FsDataProvider`, should implement [`BufferProvider`].
-//!
-//! [`BufferProvider`] can be made into a trait object. It is used over FFI.
-//!
-//! ## Serializing
-//!
-//! Providers that have full type information should implement [`DynProvider`]`<`[`SerializeMarker`]`>`.
-//! Note that a provider like `FsDataProvider` cannot implement that trait, because type information
-//! on the data structs is required in order to deserialize and then serialize them.
+//! such as `FsDataProvider`, should implement [`BufferProvider`]. These can be converted into
+//! [`DeserializingBufferProvider`] using the [`as_deserializing`](AsDeserializingBufferProvider::as_deserializing)
+//! convenience method.
 //!
-//! [`DynProvider`]`<`[`SerializeMarker`]`>` is used by data exporters such as `FilesystemExporter`.
-//!
-//! [`DynProvider`]: crate::data_provider::DynProvider
 //! [`BufferProvider`]: crate::buf::BufferProvider
 
 pub mod borrow_de_utils;
-mod de;
-#[cfg(feature = "datagen")]
-mod ser;
 
-pub use de::AsDeserializingBufferProvider;
```

Comment:
Nit: Reduce the diff size and put the code from the `ser` and `de` modules back where it came from.

Reply:
Well `ser` doesn't exist anymore, it got folded into `datagen`. I've moved `de` up because I think it makes less sense if there's no corresponding `ser` next to it (the borrow_de_utils don't justify the extra level imho)

Reply:
ok. I verified that the old de.rs and the new mod.rs have the same content.